packages/core/src/conversation/SubconversationService/SubconversationService.ts

```diff
@@ -0,0 +1,332 @@
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {SUBCONVERSATION_ID, Subconversation} from '@wireapp/api-client/lib/conversation';
+import {QualifiedId} from '@wireapp/api-client/lib/user';
+import {TimeInMillis} from '@wireapp/commons/lib/util/TimeUtil';
+import logdown from 'logdown';
+
+import {APIClient} from '@wireapp/api-client';
+import {TypedEventEmitter} from '@wireapp/commons';
+
+import {MLSService} from '../../messagingProtocols/mls';
+import {subconversationGroupIdStore} from '../../messagingProtocols/mls/MLSService/stores/subconversationGroupIdStore/subconversationGroupIdStore';
+import {constructFullyQualifiedClientId} from '../../util/fullyQualifiedClientIdUtils';
+
+type Events = {
+  MLSConversationRecovered: {conversationId: QualifiedId};
+};
+
+export interface SubconversationEpochInfoMember {
+  userid: `${string}@${string}`;
+  clientid: string;
+  in_subconv: boolean;
+}
+
+const MLS_CONVERSATION_KEY_LENGTH = 32;
+
+export class SubconversationService extends TypedEventEmitter<Events> {
+  private readonly logger = logdown('@wireapp/core/SubconversationService');
+
+  constructor(
+    private readonly apiClient: APIClient,
+    private readonly _mlsService?: MLSService,
+  ) {
+    super();
+  }
+
+  get mlsService(): MLSService {
+    if (!this._mlsService) {
+      throw new Error('MLSService was not initialised!');
+    }
+    return this._mlsService;
+  }
+
+  /**
+   * Will join or register an mls subconversation for conference calls.
+   * Will return the secret key derived from the subconversation
+   *
+   * @param conversationId Id of the parent conversation in which the call should happen
+   */
+  public async joinConferenceSubconversation(
+    conversationId: QualifiedId,
+    shouldRetry = true,
+  ): Promise<{groupId: string; epoch: number}> {
+    try {
+      const {
+        group_id: subconversationGroupId,
+        epoch: subconversationEpoch,
+        epoch_timestamp: subconversationEpochTimestamp,
+        subconv_id: subconversationId,
+      } = await this.getConferenceSubconversation(conversationId);
+
+      if (subconversationEpoch === 0) {
+        const doesConversationExistsLocally = await this.mlsService.conversationExists(subconversationGroupId);
+        if (doesConversationExistsLocally) {
+          await this.mlsService.wipeConversation(subconversationGroupId);
+        }
```

Comment:
This is the only part that was added - we need to make sure local group is wiped before we try to register it again.